src/init/features/hosting/github.ts

```diff
@@ -0,0 +1,517 @@
+import { bold } from "cli-color";
+import * as fs from "fs";
+import * as yaml from "js-yaml";
+import { safeLoad } from "js-yaml";
+import * as ora from "ora";
+import * as path from "path";
+import * as sodium from "tweetsodium";
+import { Setup } from "../..";
+import { login } from "../../../auth";
+import { dirExistsSync } from "../../../fsutils";
+import {
+  createServiceAccount,
+  createServiceAccountKey,
+  deleteServiceAccount,
+} from "../../../gcp/iam";
+import { addServiceAccountToRoles, firebaseRoles } from "../../../gcp/resourceManager";
+import * as logger from "../../../logger";
+import { prompt } from "../../../prompt";
+import { logBullet, logLabeledBullet, logSuccess } from "../../../utils";
+import api = require("../../../api");
+
+const GITHUB_DIR = `.github`;
+const WORKFLOW_DIR = `${GITHUB_DIR}/workflows`;
+
+const YML_RELATIVE_PATH_PULL_REQUEST = `${WORKFLOW_DIR}/firebase-hosting-pull-request.yml`;
+const YML_FULL_PATH_PULL_REQUEST = path.join(process.cwd(), YML_RELATIVE_PATH_PULL_REQUEST);
+
+const YML_RELATIVE_PATH_MERGE = `${WORKFLOW_DIR}/firebase-hosting-merge.yml`;
+const YML_FULL_PATH_MERGE = path.join(process.cwd(), YML_RELATIVE_PATH_MERGE);
+
+const GIT_CONFIG_PATH = path.join(process.cwd(), ".git/config");
+
+const CHECKOUT_GITHUB_ACTION_NAME = "actions/checkout@v2";
+const HOSTING_GITHUB_ACTION_NAME = "FirebaseExtended/action-hosting-deploy@v0";
+
+export async function initGitHub(setup: Setup, config: any, options: any) {
```

Comment:
Can you add a JSDoc block for this? It does quite a bit - a brief description would be helpful

Reply:
added. LGTY?